pkg/operator/options/options.go

```diff
@@ -101,6 +102,7 @@ func (o *Options) AddFlags(fs *coreoptions.FlagSet) {
 	fs.StringVar(&o.LinuxAdminUsername, "linux-admin-username", env.WithDefaultString("LINUX_ADMIN_USERNAME", "azureuser"), "The admin username for Linux VMs.")
 	fs.StringVar(&o.SSHPublicKey, "ssh-public-key", env.WithDefaultString("SSH_PUBLIC_KEY", ""), "[REQUIRED] VM SSH public key.")
 	fs.StringVar(&o.NetworkPlugin, "network-plugin", env.WithDefaultString("NETWORK_PLUGIN", consts.NetworkPluginAzure), "The network plugin used by the cluster.")
+	fs.StringVar(&o.ClusterDNSServiceIP, "dns-service-ip", env.WithDefaultString("CLUSTER_DNS_SERVICE_IP", ""), "The IP address of cluster DNS service.")
```

Comment:
I know I suggested CLUSTER_ prefix, but on a closer look we make an effort to use environment variables that match flags, and dns-service-ip matches current az aks cli and API, so I take that back.

```suggestion
	fs.StringVar(&o.ClusterDNSServiceIP, "dns-service-ip", env.WithDefaultString("DNS_SERVICE_IP", ""), "The IP address of cluster DNS service.")
```